File: project122.jl

dict = Dict()
dict[1] = 0

index = 1
array = [[1]]
while(true)
    if index > 10 break end
    arrayIndex = length(array)
    prev = 0
    for i in index:arrayIndex
        if prev == i break end
        lastElem = array[i][lastindex(array[i])] # 1
        # println("lastElem = $lastElem")
        lenX = length(array[i])
        # println("array = $array")
        # println("index = $i")
        # println("array[i] = $(array[i])")
        for x in 1:lenX
            temp = copy(array[i]) # [1]
            term = array[i][x] + lastElem
            push!(temp,term)
            push!(array,temp)
            try
                if dict[term] > index
                    dict[term] = index
                end
            catch e
                if isa(e,KeyError)
                    dict[term] = index
                end
            end
        end
        prev = i
    end
    index += 1
end

ans = 0
for i in 1:200
    ans += dict[i]
end

println(ans)

File: project122.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int index = 0;
    map<int,int> dict;
    vector<vector<int>> array = {{1}};
    int n = 2;
    while(true) {
        if(n > 200) break;
        int lastElem = array[index].back();
        int lenx = array[index].size();
        for(int x=0;x<lenx;x++) {
            vector<int> temp = array[index];
            int term = array[index][x] + lastElem;
            temp.push_back(term);
            array.push_back(temp);
            if(dict[term] == 0) {
                dict[term] = lenx;
            }
            while(true) {
                if(n>200) break;
                if(dict[n] != 0) {
                    n++;
                }else if(dict[n] == 0) {
                    break;
                }
            }
        }
        index += 1;
    }
    int ans = 0;
    for(int i=2;i<=200;i++) {
        ans += dict[i];
    }
    cout << "ans = " << ans << '\n';
    
}